filebeat/harvester/log.go

```diff
@@ -104,24 +114,50 @@ func (h *Harvester) Harvest() {
 		// Reset Backoff
 		h.backoff = h.Config.BackoffDuration
 
-		// Sends text to spooler
-		event := &input.FileEvent{
-			ReadTime:     lastReadTime,
-			Source:       &h.Path,
-			InputType:    h.Config.InputType,
-			DocumentType: h.Config.DocumentType,
-			Offset:       h.Offset,
-			Bytes:        bytesRead,
-			Text:         &text,
-			Fields:       &h.Config.Fields,
-			Fileinfo:     &info,
+		if h.shouldExportLine(text) {
+
+			// Sends text to spooler
+			event := &input.FileEvent{
+				ReadTime:     lastReadTime,
+				Source:       &h.Path,
+				InputType:    h.Config.InputType,
+				DocumentType: h.Config.DocumentType,
+				Offset:       h.Offset,
+				Bytes:        bytesRead,
+				Text:         &text,
+				Fields:       &h.Config.Fields,
+				Fileinfo:     &info,
+			}
+
+			event.SetFieldsUnderRoot(h.Config.FieldsUnderRoot)
+			h.SpoolerChan <- event // ship the new event downstream
 		}
 
+		// Set Offset
 		h.Offset += int64(bytesRead) // Update offset if complete line has been processed
+	}
+}
 
-		event.SetFieldsUnderRoot(h.Config.FieldsUnderRoot)
-		h.SpoolerChan <- event // ship the new event downstream
+// shouldExportLine decides if the line is exported or not based on
+// the include_lines and exclude_lines options.
+func (h *Harvester) shouldExportLine(line string) bool {
+	if len(h.IncludeLinesRegexp) > 0 {
+		if !MatchAnyRegexps(h.IncludeLinesRegexp, line) {
+			// drop line
+			logp.Debug("drop", "Drop line %s", line)
```

Comment:
I would suggest to use here "harvester" instead of drop to have it under the same selector as all other harvester messages. And the drop line I would use: Line dropped because does not match include pattern: %s. Below the same: Line dropped because does match exclude pattern: %s This makes it easier to see where the message is coming from.